code/modules/vehicles/cars/golfcart/golfcart.dm

```diff
@@ -0,0 +1,677 @@
+			person.apply_damage(2 * damage, BRUTE, BODY_ZONE_CHEST)
+			person.apply_damage(0.5 * damage, BRUTE, BODY_ZONE_L_LEG)
+			person.apply_damage(0.5 * damage, BRUTE, BODY_ZONE_R_LEG)
+			person.apply_damage(0.5 * damage, BRUTE, BODY_ZONE_L_ARM)
+			person.apply_damage(0.5 * damage, BRUTE, BODY_ZONE_R_ARM)
+
+			add_mob_blood(person)
+			var/turf/below_us = get_turf(src)
+			below_us.add_mob_blood(person)
+
+			AddComponent(/datum/component/blood_walk, \
+				blood_type = /obj/effect/decal/cleanable/blood/tracks, \
+				target_dir_change = TRUE, \
+				transfer_blood_dna = TRUE, \
+				max_blood = 4)
+
+/obj/vehicle/ridden/golfcart/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if (!is_hotrod())
+		return
+	for(var/mob/living/future_pancake in loc)
+		run_over(future_pancake)
+
+/obj/vehicle/ridden/golfcart/item_interaction(mob/living/user, obj/item/attacking_item, list/modifiers)
+	if (!hood_open)
+		return ..()
+	if (istype(attacking_item, /obj/item/v8_engine))
+		if (engine || cell)
+			balloon_alert(user, "already has an engine!")
+			return ITEM_INTERACT_BLOCKING
+		user.transferItemToLoc(attacking_item, src)
+		engine = attacking_item
+		engine_state = ENGINE_UNWRENCHED
+		balloon_alert(user, "installed \the [engine]")
+		return ITEM_INTERACT_SUCCESS
+	if (istype(attacking_item, /obj/item/stock_parts/power_store/cell))
+		if (cell || engine)
+			balloon_alert(user, "already has an engine!")
+			return ITEM_INTERACT_BLOCKING
+		user.transferItemToLoc(attacking_item, src)
+		cell = attacking_item
+		balloon_alert(user, "installed \the [cell]")
+		return ITEM_INTERACT_SUCCESS
+	return ..()
+
+/obj/vehicle/ridden/golfcart/attack_hand(mob/living/user, list/modifiers)
+	if (!hood_open)
+		return ..()
+	if (engine && engine_state == ENGINE_UNWRENCHED)
+		. = TRUE
+		var/obj/item/engine_item = engine
+		engine_state = null
+		engine = null
+		if (user.put_in_hands(engine_item))
+			return
+		engine_item.forceMove(drop_location())
+		return
+	if (isnull(cell))
+		return ..()
+	var/obj/item/stock_parts/power_store/cell/cell_to_take = cell
+	cell = null
+	. = TRUE
+	if (user.put_in_hands(cell_to_take))
+		return
+	cell_to_take.forceMove(drop_location())
+
+/obj/vehicle/ridden/golfcart/proc/can_wrench_engine()
+	return hood_open && engine && (engine_state == ENGINE_UNWRENCHED || engine_state == ENGINE_WRENCHED)
+
+/obj/vehicle/ridden/golfcart/proc/can_weld_engine()
+	return hood_open && engine && (engine_state == ENGINE_WRENCHED || engine_state == ENGINE_WELDED)
+
+/obj/vehicle/ridden/golfcart/proc/set_to_hotrod_sprite()
+	if (icon == 'icons/obj/toys/golfcart_hotrod_split.dmi')
+		return
+	icon = 'icons/obj/toys/golfcart_hotrod_split.dmi'
+	update_appearance(UPDATE_ICON)
+
+/obj/vehicle/ridden/golfcart/proc/set_to_default_sprite()
+	if (icon == 'icons/obj/toys/golfcart_split.dmi')
+		return
+	icon = 'icons/obj/toys/golfcart_split.dmi'
+	update_appearance(UPDATE_ICON)
```

Comment:
Merge these two procs behaviors into `set_engine_state`, move the update_apperance(UPDATE_ICON) to the end of `set_engine_state`, and that'll save you some proc call overhead